### DISCUSSION TITLE
Question about the correct use of the Field method

### DISCUSSION THREAD

**Discussion by mxustin** [pydantic/pydantic#7553] [Question] [answered]
Hi, there!

At https://docs.pydantic.dev/2.2/usage/fields/ we can read:

`id: int = Field(default_factory=lambda: uuid4().hex)`

However, I would like to ask, is this a good example of how to use it? an indication of the type of integers coming immediately after the field name in this case may be misleading. 

The fact is that the id field of the User class will eventually have the type str. And such a class declaration, as shown in the example, may make someone think that there should be an integer there, since int is written. I'll show you by example what I mean.

When I read such code:

```
class User(BaseModel):
    id: int = Field(default_factory=lambda: uuid4().hex)


u = User()
print(u.id)
print(type(u.id))
```

...I can decide that the result will be <class 'int'>. But, in reality...

```
045fac9ae5334d9691b070a60e353065
<class 'str'>

```

Perhaps it makes sense to assume that a more correct example would be:

```
class User(BaseModel):
    id: int = Field(default_factory=lambda: uuid4().int)


u = User()
print(u.id)
print(type(u.id))
```

it returns:

```
3369268732298293168957280164202434687
<class 'int'>
```

... or maybe:

```
class User(BaseModel):
    id: str = Field(default_factory=lambda: uuid4().hex)


u = User()
print(u.id)
print(type(u.id))
```

it returns:

```
a864729fb6b4434c9a766abed70cc8c4
<class 'str'>
```

**Comment by Viicos** [comment, marked as answer]
Fixed by https://github.com/pydantic/pydantic/pull/7311, you can see the result in the [dev docs](https://docs.pydantic.dev/dev-v2/usage/fields/#default-values).

**Reply by mxustin** [reply]
10q
